server/valgrind-out.txt

```diff
@@ -0,0 +1,136 @@
+--16528-- REDIR: 0x492f4f0 (libc.so.6:wcscpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912250 (libc.so.6:strcat) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913be0 (libc.so.6:strncasecmp_l) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913500 (libc.so.6:bcmp) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x491b5f0 (libc.so.6:memrchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4915090 (libc.so.6:strchrnul) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912410 (libc.so.6:strcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913b90 (libc.so.6:strcasecmp_l) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912720 (libc.so.6:strlen) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912950 (libc.so.6:strncpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x49139d0 (libc.so.6:stpcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913470 (libc.so.6:memchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912b80 (libc.so.6:strspn) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913850 (libc.so.6:mempcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4913b40 (libc.so.6:strncasecmp) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4915000 (libc.so.6:rawmemchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4912820 (libc.so.6:strncat) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--16528-- REDIR: 0x4a07790 (libc.so.6:__strrchr_avx2) redirected to 0x484e810 (rindex)
+--16528-- REDIR: 0x4a0b100 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x4852770 (memset)
+--16528-- REDIR: 0x490f120 (libc.so.6:malloc) redirected to 0x4848820 (malloc)
+--16528-- REDIR: 0x490f460 (libc.so.6:free) redirected to 0x484b210 (free)
+--16528-- REDIR: 0x49105a0 (libc.so.6:calloc) redirected to 0x484d9d0 (calloc)
+--16528-- REDIR: 0x4a07580 (libc.so.6:__strchrnul_avx2) redirected to 0x4853330 (strchrnul)
+--16528-- REDIR: 0x4a07960 (libc.so.6:__strlen_avx2) redirected to 0x484ed10 (strlen)
+--16528-- REDIR: 0x4a0a940 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x4852880 (memmove)
+--16528-- REDIR: 0x4913040 (libc.so.6:__GI_strstr) redirected to 0x4853ae0 (__strstr_sse2)
+--16528-- REDIR: 0x4a03c00 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x4852010 (bcmp)
+--16528-- REDIR: 0x4a02ac0 (libc.so.6:__strcmp_avx2) redirected to 0x484fbd0 (strcmp)
+--16528-- REDIR: 0x490f7c0 (libc.so.6:realloc) redirected to 0x484dc50 (realloc)
+--16528-- REDIR: 0x4a08420 (libc.so.6:__strncat_avx2) redirected to 0x484ebc0 (strncat)
+--16528-- REDIR: 0x4a0a900 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4853440 (mempcpy)
+==16528== Warning: invalid file descriptor -1 in syscall close()
```

Comment:
This is highlighting a real issue - you need to check for valid handle before closing in your signal handler.  It will be difficult to free these resources safely with the exit from signal handler approach.

Reply:
Thank you for pointing it out. I have fixed this with A6 release.